### PR TITLE
improvement(remove default from get method): Remove 'default' parameter from params.get() method

### PR DESCRIPTION
**Pull request by juliayakovlev**
Because of most of defaults are stored in the default yamls
and to prevent adding defaults in code it's better to remove
this parameter.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/sct_config.py [hunk 1]
```diff
@@ -1087,12 +1087,12 @@ def __init__(self):
...
         anyconfig.merge(self, files)
 
-        regions_data = self.get('regions_data', {})
+        regions_data = self.get('regions_data') or {}
         if regions_data:
             del self['regions_data']
 
         # 2.2) load the region data
-        region_names = self.get('region_name', '').split()
+        region_names = self.get('region_name') or ''.split()
```
Comment (bentsi) [inline sdcm/sct_config.py:1095]: need to run .split on the results of `self.get('region_name')`

Comment (juliayakovlev) [reply sdcm/sct_config.py]: fixed

File: longevity_large_partition_test.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 class LargePartitionLongevityTest(LongevityTest):
 
     def test_large_partition_longevity(self):
-        compaction_strategy = self.params.get('compaction_strategy', default=CompactionStrategy.SIZE_TIERED.value)
+        compaction_strategy = self.params.get('compaction_strategy')
```
Comment (dkropachev) [inline longevity_large_partition_test.py:9]: You removed it, but did not go thru all test files that are using it, to see if there was no compaction strategy defined.

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: longevity_large_partition_test.py (hunk 2 above)
Comment (juliayakovlev) [reply longevity_large_partition_test.py:9]: @dkropachev 
I am not sure that I understand correctly this comment
Default compaction_strategy is defined in the test_defaults.py as SIZE_TIERED.

File: sdcm/sct_config.py [hunk 3]
```diff
@@ -1151,8 +1144,8 @@ def __init__(self):
                             self[key] += " {}".format(value)
 
         # 5) handle scylla_version if exists
-        scylla_version = self.get('scylla_version', None)
-        scylla_linux_distro = self.get('scylla_linux_distro', '')
+        scylla_version = self.get('scylla_version')
+        scylla_linux_distro = self.get('scylla_linux_distro') or ''
```
Comment (bentsi) [inline sdcm/sct_config.py:1148]: remove '', the default should be in the defaults yaml

File: sdcm/sct_config.py [hunk 4]
```diff
@@ -1288,18 +1281,15 @@ def _update_environment_variables(self, replace=False):
             if opt["name"] in self and (opt["env"] not in os.environ or replace):
                 os.environ[opt["env"]] = str(self[opt["name"]])
 
-    def get(self, key, default=UnsetMarker):
+    def get(self, key):
         """
         get a specific configuration by key
         :param key: the key to get
         :param default: default value if the key doesn't exist
```
Comment (bentsi) [inline sdcm/sct_config.py:1288]: remove docstring

Comment (bentsi) [reply sdcm/sct_config.py:1288]: add type anntations

**Review by bentsi — COMMENTED** [review]
please go over all the changes that include `self.get(...) or ""`

File: sdcm/sct_config.py (hunk 4 above)
Comment (juliayakovlev) [reply sdcm/sct_config.py]: changed

File: sdcm/sct_config.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/sct_config.py]: done

**Comment by bentsi** [comment]
@juliayakovlev please resolve conflicts

File: sdcm/nemesis.py [hunk 5]
```diff
@@ -801,7 +801,7 @@ def call_random_disrupt_method(self, disrupt_methods=None, predefined_sequence=F
                 # Adjust the rate according to the test duration. Try to call more unique
                 # methods and don't wait to long time to meet the balance if the test
                 # duration is short.
-                test_duration = self.cluster.params.get('test_duration', default=180)
+                test_duration = self.cluster.params.get('test_duration') or 180
```
Comment (bentsi) [inline sdcm/nemesis.py:804]: any reason to keep the default here?

File: performance_regression_test.py [hunk 6]
```diff
@@ -188,7 +188,7 @@ def run_read_workload(self, nemesis=False):
         self.create_test_stats(sub_type='read', doc_id_with_timestamp=True)
         stress_queue = self.run_stress_thread(stress_cmd=base_cmd_r, stress_num=1, stats_aggregate_cmds=False)
         if nemesis:
-            interval = self.params.get('nemesis_interval', default=10)
+            interval = self.params.get('nemesis_interval') or 10
```
Comment (bentsi) [inline performance_regression_test.py:191]: no need to keep the defaults here

File: performance_regression_test.py [hunk 7]
```diff
@@ -205,7 +205,7 @@ def run_write_workload(self, nemesis=False):
         stress_queue = self.run_stress_thread(stress_cmd=base_cmd_w, stress_num=1, stats_aggregate_cmds=False)
         if nemesis:
             self.db_cluster.add_nemesis(nemesis=self.get_nemesis_class(), tester_obj=self)
-            self.db_cluster.start_nemesis(interval=self.params.get('nemesis_interval', default=10))
+            self.db_cluster.start_nemesis(interval=self.params.get('nemesis_interval') or 10)
```
Comment (bentsi) [inline performance_regression_test.py:208]: no need to keep the defaults here

File: performance_regression_test.py [hunk 8]
```diff
@@ -219,7 +219,7 @@ def run_mixed_workload(self, nemesis=False):
         stress_queue = self.run_stress_thread(stress_cmd=base_cmd_m, stress_num=1, stats_aggregate_cmds=False)
         if nemesis:
             self.db_cluster.add_nemesis(nemesis=self.get_nemesis_class(), tester_obj=self)
-            self.db_cluster.start_nemesis(interval=self.params.get('nemesis_interval', default=10))
+            self.db_cluster.start_nemesis(interval=self.params.get('nemesis_interval') or 10)
```
Comment (bentsi) [inline performance_regression_test.py:222]: no need to keep the defaults here

File: performance_regression_test.py [hunk 9]
```diff
@@ -232,7 +232,7 @@ def run_workload(self, stress_cmd, nemesis=False):
         self.create_test_stats(sub_type=sub_type, doc_id_with_timestamp=True)
         stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=1, stats_aggregate_cmds=False)
         if nemesis:
-            interval = self.params.get('nemesis_interval', default=15)
+            interval = self.params.get('nemesis_interval') or 15
```
Comment (bentsi) [inline performance_regression_test.py:235]: no need to keep the defaults here

File: grow_cluster_test.py [hunk 10]
```diff
@@ -91,7 +91,7 @@ def grow_cluster(self, cluster_target_size, stress_cmd):
         start = datetime.datetime.now()
         self.log.info('Starting to grow cluster: %s', str(start))
 
-        add_node_cnt = self.params.get('add_node_cnt', default=1)
+        add_node_cnt = self.params.get('add_node_cnt') or 1
```
Comment (bentsi) [inline grow_cluster_test.py:94]: why do you keep defaults here?

File: grow_cluster_test.py [hunk 11]
```diff
@@ -60,10 +60,10 @@ def get_stress_cmd(self, mode='write', duration=None):
         :rtype: basestring
         """
         ip = self.db_cluster.get_node_private_ips()[0]
-        population_size = self.params.get('cassandra_stress_population_size', default=1000000)
+        population_size = self.params.get('cassandra_stress_population_size') or 1000000
```
Comment (bentsi) [inline grow_cluster_test.py:63]: why do you keep defaults? it is set in `test-cases/features/scale-cluster.yaml`

File: grow_cluster_test.py [hunk 12]
```diff
@@ -60,10 +60,10 @@ def get_stress_cmd(self, mode='write', duration=None):
...
         if not duration:
-            duration = self.params.get('test_duration', default=60)
-        threads = self.params.get('cassandra_stress_threads', default=1000)
+            duration = self.params.get('test_duration')
+        threads = self.params.get('cassandra_stress_threads') or 1000
```
Comment (bentsi) [inline grow_cluster_test.py:66]: why do you keep defaults? it is set in `test-cases/features/scale-cluster.yaml`

File: longevity_test.py [hunk 13]
```diff
@@ -306,15 +306,15 @@ def test_user_batch_custom_time(self):
...
 
-        if not self.params.get('reuse_cluster', default=False):
+        if not self.params.get('reuse_cluster'):
             self._pre_create_templated_user_schema()
 
             # Start new nodes
             # we are starting this test case with only one db to make creating of the tables quicker
             # gossip with multiple node cluster make this painfully slower
-            add_node_cnt = self.params.get('add_node_cnt', default=1)
+            add_node_cnt = self.params.get('add_node_cnt') or 1
```
Comment (bentsi) [inline longevity_test.py:317]: add to `test_defaults.yaml`, no need to keep the default value here

File: sdcm/cluster.py [hunk 14]
```diff
@@ -2831,8 +2831,8 @@ def terminate_node(self, node):
         node.destroy()
 
     def get_db_auth(self):
-        user = self.params.get('authenticator_user', default=None)
-        password = self.params.get('authenticator_password', default=None)
+        user = self.params.get('authenticator_user')
```
Comment (bentsi) [inline sdcm/cluster.py:2834]: set the default in `test_default.yaml`

File: sdcm/cluster.py [hunk 15]
```diff
@@ -2872,16 +2872,16 @@ def _create_session(self, node, keyspace, user, password, compression,
             protocol_version = 3
 
         authenticator = self.params.get('authenticator')
-        if user is None and password is None and (authenticator and authenticator == 'PasswordAuthenticator'):
-            user = self.params.get('authenticator_user', default='cassandra')
-            password = self.params.get('authenticator_password', default='cassandra')
+        if authenticator and authenticator == 'PasswordAuthenticator':
+            user = self.params.get('authenticator_user')
+            password = self.params.get('authenticator_password')
```
Comment (bentsi) [inline sdcm/cluster.py:2877]: set the default in `test_default.yaml`

File: sdcm/cluster.py [hunk 16]
```diff
@@ -3660,8 +3660,8 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
             self.log.info('io.conf right after reboot')
             node.remoter.sudo('cat /etc/scylla.d/io.conf')
 
-            if self.params.get('use_mgmt', None):
-                pkgs_url = self.params.get('scylla_mgmt_pkg', None)
+            if self.params.get('use_mgmt'):
```
Comment (bentsi) [inline sdcm/cluster.py:3663]: set default to false in `defaults/test_default.yaml`

File: sdcm/cluster.py [hunk 17]
```diff
@@ -4507,7 +4507,7 @@ def reconfigure_scylla_monitoring(self):
                 ''')
                 node.remoter.run("sudo bash -ce '%s'" % configure_self_node_exporter, verbose=True)
 
-            if self.params.get('cloud_prom_bearer_token', None):
+            if self.params.get('cloud_prom_bearer_token'):
```
Comment (bentsi) [inline sdcm/cluster.py:4510]: add to test_defautls.yaml

File: sdcm/gemini_thread.py [hunk 18]
```diff
@@ -72,7 +72,7 @@ def gemini_result_file(self):
         return self._gemini_result_file
 
     def _generate_gemini_command(self):
-        seed = self.params.get('gemini_seed', None)
+        seed = self.params.get('gemini_seed')
```
Comment (bentsi) [inline sdcm/gemini_thread.py:75]: add to `test_default.yaml`

File: sdcm/sct_config.py [hunk 19]
```diff
@@ -1140,8 +1133,8 @@ def __init__(self):
 
         # 4) assume multi dc by n_db_nodes set size
         if 'aws' in cluster_backend:
-            num_of_regions = len(self.get('region_name', '').split())
-            num_of_db_nodes_sets = len(str(self.get('n_db_nodes', '')).split(' '))
+            num_of_regions = len((self.get('region_name') or '').split())
+            num_of_db_nodes_sets = len(str(self.get('n_db_nodes') or '').split(' '))
```
Comment (bentsi) [inline sdcm/sct_config.py:1137]: `n_db_nodes` is mandatory, need to fail the test. Put an assert if it is empty or 0

File: sdcm/sct_config.py [hunk 20]
```diff
@@ -1444,9 +1431,9 @@ def _check_backend_defaults(backend, required_params):
                 raise ValueError(error_message_template.format('<max PK value> should be bigger then <min PK value>. '))
 
         # verify that the AMIs used all have 'user_data_format_version' tag
-        ami_id_db_scylla = self.get('ami_id_db_scylla', default='').split()
-        region_names = self.get('region_name', default='').split()
-        ami_id_db_oracle = self.get('ami_id_db_oracle', default='').split()
+        ami_id_db_scylla = (self.get('ami_id_db_scylla') or '').split()
```
Comment (bentsi) [inline sdcm/sct_config.py:1434]: 2) put the `ami_id_db_scylla: ''` and `ami_id_db_oracle: ''` to aws_defaults.yaml
3) remove or ' '
4) no need for default for `region_name` it is checked before

File: sdcm/utils/data_validator.py [hunk 21]
```diff
@@ -167,7 +167,7 @@ def __init__(self, longevity_self_object, user_profile_name, base_table_partitio
     @property
     def keyspace_name(self):
         if not self._keyspace_name:
-            prepare_write_cmd = self.longevity_self_object.params.get(self.stress_cmds_part, default=[])
+            prepare_write_cmd = self.longevity_self_object.params.get(self.stress_cmds_part) or []
```
Comment (bentsi) [inline sdcm/utils/data_validator.py:170]: why do you keep default here?

File: sdcm/utils/data_validator.py [hunk 22]
```diff
@@ -251,7 +251,7 @@ def get_view_name_from_profile(self, name_substr, cs_profile=None, all_entries=F
         mv_names = []
 
         if not cs_profile:
-            stress_cmd = self.longevity_self_object.params.get(self.stress_cmds_part, default=[])
+            stress_cmd = self.longevity_self_object.params.get(self.stress_cmds_part) or []
```
Comment (bentsi) [inline sdcm/utils/data_validator.py:254]: why do you keep default here?

File: grow_cluster_test.py (hunk 11 above)
Comment (juliayakovlev) [reply grow_cluster_test.py:63]: we have no default for that. If another yaml will be added to use this test and this parameter will be missed there, do we wnat this test will fail?

File: sdcm/cluster.py (hunk 14 above)
Comment (juliayakovlev) [reply sdcm/cluster.py:2834]: I don't thins it's correct. We don't want to use credentials in every test

File: sdcm/gemini_thread.py (hunk 18 above)
Comment (juliayakovlev) [reply sdcm/gemini_thread.py:75]: a you sure we need it there? By test code seems we don't need the default value for this parameter

File: sdcm/sct_config.py (hunk 19 above)
Comment (juliayakovlev) [reply sdcm/sct_config.py:1137]: it's part of required params and assert with all other requirement params

File: sdcm/utils/data_validator.py (hunk 22 above)
Comment (juliayakovlev) [reply sdcm/utils/data_validator.py:254]: Expected list there. If there is no stress commands in this part, empty list will be at least

File: grow_cluster_test.py (hunk 12 above)
Comment (juliayakovlev) [reply grow_cluster_test.py:66]: same as above

File: grow_cluster_test.py (hunk 10 above)
Comment (juliayakovlev) [reply grow_cluster_test.py:94]: removed

File: longevity_test.py (hunk 13 above)
Comment (juliayakovlev) [reply longevity_test.py:317]: done

File: performance_regression_test.py (hunk 6 above)
Comment (juliayakovlev) [reply performance_regression_test.py:191]: removed

File: performance_regression_test.py (hunk 7 above)
Comment (juliayakovlev) [reply performance_regression_test.py:208]: removed

File: performance_regression_test.py (hunk 9 above)
Comment (juliayakovlev) [reply performance_regression_test.py:235]: removed

File: sdcm/cluster.py (hunk 15 above)
Comment (juliayakovlev) [reply sdcm/cluster.py:2877]: Do we want to use credentials in every stress command?

File: sdcm/cluster.py (hunk 17 above)
Comment (juliayakovlev) [reply sdcm/cluster.py:4510]: done

File: sdcm/nemesis.py (hunk 5 above)
Comment (juliayakovlev) [reply sdcm/nemesis.py:804]: removed

File: sdcm/sct_config.py (hunk 19 above)
Comment (juliayakovlev) [reply sdcm/sct_config.py:1137]: removed "or ''"

File: sdcm/sct_config.py (hunk 20 above)
Comment (juliayakovlev) [reply sdcm/sct_config.py:1434]: done

File: sdcm/utils/data_validator.py (hunk 21 above)
Comment (juliayakovlev) [reply sdcm/utils/data_validator.py:170]: Expected list there. If there is no stress commands in this part, empty list will be at least

File: grow_cluster_test.py (hunk 11 above)
Comment (bentsi) [reply grow_cluster_test.py:63]: that will be the default

File: sdcm/tester.py [hunk 23]
```diff
@@ -243,7 +243,7 @@ def __init__(self, *args):  # pylint: disable=too-many-statements
         Setup.keep_cluster(node_type='loader_nodes', val=post_behavior_loader_nodes)
 
         set_cluster_duration(self._duration)
-        cluster_backend = self.params.get('cluster_backend', default='')
+        cluster_backend = self.params.get('cluster_backend') or ''
```
Comment (bentsi) [inline sdcm/tester.py:246]: remove

File: sdcm/tester.py [hunk 24]
```diff
@@ -562,12 +562,12 @@ def get_cluster_gce(self, loader_info, db_info, monitor_info):
         gce_image_monitor = self.params.get('gce_image_monitor')
         if not gce_image_monitor:
             gce_image_monitor = self.params.get('gce_image')
-        cluster_additional_disks = {'pd-ssd': self.params.get('gce_pd_ssd_disk_size_db', default=0),
-                                    'pd-standard': self.params.get('gce_pd_standard_disk_size_db', default=0)}
+        cluster_additional_disks = {'pd-ssd': self.params.get('gce_pd_ssd_disk_size_db') or 0,
```
Comment (bentsi) [inline sdcm/tester.py:565]: add to SCT config and to GCE defaults

File: sdcm/cluster.py (hunk 14 above)
Comment (bentsi) [reply sdcm/cluster.py:2846]: no set to "" (empty string)

File: sdcm/tester.py [hunk 25]
```diff
@@ -402,7 +402,7 @@ def setUp(self):
         self.update_certificates()
 
         # download rpms for update_db_packages
-        update_db_packages = self.params.get('update_db_packages', default=None)
+        update_db_packages = self.params.get('update_db_packages')
```
Comment (bentsi) [inline sdcm/tester.py:405]: add default

File: sdcm/nemesis.py [hunk 26]
```diff
@@ -2366,7 +2365,7 @@ def disrupt_hot_reloading_internode_certificate(self):
         This test will create and reload new certificates for the inter node communication.
         '''
         self._set_current_disruption('ServerSslHotReloadingNemesis')
-        if not self.cluster.params.get('server_encrypt', None):
+        if not self.cluster.params.get('server_encrypt'):
```
Comment (bentsi) [inline sdcm/nemesis.py:2368]: add to defaults

**Review by bentsi — APPROVED** [review]

**Pull request merged**